contrib/command-not-found/pk-command-not-found.c

```diff
@@ -623,21 +623,24 @@ pk_cnf_get_config (void)
 }
 
 static gint
-pk_cnf_spawn_command (const gchar *exec, gchar **arguments)
+pk_cnf_spawn_command (const gchar *exec, gchar **arguments, guint arguments_count)
 {
+	g_autofree gchar **argv = NULL;
 	gint exit_status = EXIT_FAILURE;
 	g_autoptr(GError) error = NULL;
-	g_autofree gchar *args = NULL;
-	g_autofree gchar *cmd = NULL;
 
 	/* ensure program starts on a fresh line */
 	g_print ("\n");
 
-	args = g_strjoinv (" ", arguments);
-	cmd = g_strjoin (" ", exec, args, NULL);
-	if (!g_spawn_command_line_sync (cmd, NULL, NULL, &exit_status, &error)) {
+	argv = g_new0 (gchar *, arguments_count + 2);
+	argv[0] = (gchar*) exec;
+  for (guint i = 0; i <= arguments_count; i++) {
+    argv[1+i] = arguments[i];
+  }
```

Comment:
tabs here, and then rebase all the patches to one commit and force push to the branch please.